manifests/ulimit.pp

```diff
@@ -17,6 +17,13 @@
 class redis::ulimit {
   assert_private('The redis::ulimit class is only to be called from the redis::config class')
 
+  # Only necessary for Puppet < 6.1.0,
```

Comment:
This should be inside the `if $service_provider_lookup == 'systemd' {`. Otherwise it's included on non-systemd distros. I'd also write `Augeas['Systemd redis ulimit'] ~> Class['systemd::systemctl::daemon_reload']` instead of using `$_notify`

Reply:
Sure, will also add this condition to the test so it does not fail anymore.